src/main/java/co/cask/gcp/speech/SpeechToTextTransform.java

```diff
@@ -0,0 +1,332 @@
+    return getSchema(entries.iterator().next().getValue(), request.getTranscriptionPartsField(),
+                     request.getTranscriptionTextField());
+  }
+
+  private SpeechSettings getSettings() throws IOException {
+    SpeechSettings.Builder builder = SpeechSettings.newBuilder();
+    if (config.getServiceAccountFilePath() != null) {
+      builder.setCredentialsProvider(() -> GCPUtils.loadServiceAccountCredentials(config.getServiceAccountFilePath()));
+    }
+    return builder.build();
+  }
+
+  private void setRecognitionConfig() {
+    recognitionConfig = RecognitionConfig.newBuilder()
+      .setEncoding(getAudioEncoding())
+      .setSampleRateHertz(Integer.parseInt(config.sampleRate))
+      .setProfanityFilter(config.profanity.equals("true"))
+      .setLanguageCode(config.language == null ? "en-US" : config.language)
+      .build();
+  }
+
+  private RecognitionConfig.AudioEncoding getAudioEncoding() {
+    RecognitionConfig.AudioEncoding encoding = RecognitionConfig.AudioEncoding.LINEAR16;
+    if (config.encoding.equalsIgnoreCase("amr")) {
+      encoding = RecognitionConfig.AudioEncoding.AMR;
+    } else if (config.encoding.equalsIgnoreCase("amr_wb")) {
+      encoding = RecognitionConfig.AudioEncoding.AMR_WB;
+    } else if (config.encoding.equalsIgnoreCase("flac")) {
+      encoding = RecognitionConfig.AudioEncoding.FLAC;
+    } else if (config.encoding.equalsIgnoreCase("mulaw")) {
+      encoding = RecognitionConfig.AudioEncoding.MULAW;
+    } else if (config.encoding.equalsIgnoreCase("ogg_opus")) {
+      encoding = RecognitionConfig.AudioEncoding.OGG_OPUS;
+    }
+    return encoding;
+  }
+
+  private void addTranscriptWithConfidence(Schema recordSchema, List<StructuredRecord> speechArray,
+                                           List<SpeechRecognitionAlternative> alternatives) {
+    for (SpeechRecognitionAlternative alternative : alternatives) {
+      float confidence = alternative.getConfidence();
+      String transcript = alternative.getTranscript();
+      StructuredRecord.Builder speech =
+        StructuredRecord.builder(recordSchema.getField(config.getTranscriptionPartsField())
+                                   .getSchema().getNonNullable().getComponentSchema());
+      speech.set("confidence", confidence);
+      speech.set("transcript", transcript);
+      speechArray.add(speech.build());
+    }
+  }
+
+  private Schema.Field transcriptWithConfidenceSchema(String transcriptPartsField) {
+    List<Schema.Field> fields = new ArrayList<>();
+    fields.add(Schema.Field.of("transcript", Schema.nullableOf(Schema.of(Schema.Type.STRING))));
+    fields.add(Schema.Field.of("confidence", Schema.nullableOf(Schema.of(Schema.Type.FLOAT))));
+    return Schema.Field.of(transcriptPartsField, Schema.nullableOf(Schema.arrayOf(Schema.recordOf("speech", fields))));
+  }
+
+  private Schema.Field topConfidenceSchema(String transcriptTextField) {
```

Comment:
there's no need for this one line method, just create the field directly. This just adds an unneeded layer of indirection.